impl/client/src/main/javascript/web/util/_a11y.js

```diff
@@ -35,6 +35,421 @@ define(["./_focus"], function(focusUtil) {
+
+    /**
+     * Helper function to check if a number is within a range; no side effects.
+     *
+     * @param index
+     * @param start
+     * @param end
+     * @returns {boolean}
+     */
+    function checkInRange(index, start, end) {
+      var rangeStart = start < end ? start : end;
+      var rangeEnd = start < end ? end : start;
+
+      return index >= rangeStart && index <= rangeEnd;
+    }
+
+    /**
+     * Select a range of options
+     *
+     * @param start
+     * @param end
+     */
+    function selectRange(start, end) {
+      // get start/end indices
+      var allOptions = listboxNode.querySelectorAll('[role="option"]');
+      var startIndex =
+          typeof start === 'number' ? start : getElementIndex(start, allOptions);
+      var endIndex =
+          typeof end === 'number' ? end : getElementIndex(end, allOptions);
+
+      for (var index = 0; index < allOptions.length; index++) {
+        var selected = checkInRange(index, startIndex, endIndex);
+        allOptions[index].setAttribute('aria-selected', selected + '');
+        if (selected) {
+          allOptions[index].classList.add('selected');
+        } else {
+          allOptions[index].classList.remove('selected');
+        }
+      }
+    }
+
+    /**
+     * @description
+     *  Focus on the first option
+     */
+    function focusFirstItem() {
+      var firstItem = listboxNode.querySelector('[role="option"]');
+
+      if (firstItem) {
+        focusItem(firstItem);
+      }
+    }
+
+    /**
+     * @description
+     *  Focus on the last option
+     */
+    function focusLastItem() {
+      var itemList = listboxNode.querySelectorAll('[role="option"]');
+
+      if (itemList.length) {
+        focusItem(itemList[itemList.length - 1]);
+      }
+    }
+
+    /**
+     * @description
+     *  Focus on the specified item
+     * @param element
+     *  The element to focus
+     */
+    function focusItem(element) {
+      defocusItem(document.getElementById(activeDescendant));
+      if (!multiselectable) {
+        element.setAttribute('aria-selected', 'true');
+      }
+      element.classList.add('keyboard-navigation-focus');
+      listboxNode.setAttribute('aria-activedescendant', element.id);
+      activeDescendant = element.id;
+
+      handleFocusChange(element);
+    }
+
+    /**
+     * @description
+     *  Defocus the specified item
+     * @param element
+     *  The element to defocus
+     */
+    function defocusItem(element) {
+      if (!element) {
+        return;
+      }
+      if (!multiselectable) {
+        element.removeAttribute('aria-selected');
+      }
+      element.classList.remove('keyboard-navigation-focus');
+    }
+
+    function clearKeysSoFarAfterDelay() {
+      if (keyClear) {
+        clearTimeout(keyClear);
+        keyClear = null;
+      }
+      var keyClear = setTimeout(
+          function () {
+            keysSoFar = '';
+            keyClear = null;
+          }.bind(this),
+          500
+      );
```

Comment:
As implemented, `keyClear` is a local variable of the `clearKeysSoFarAfterDelay` fucntion.
When entering the function, its value will **always** be `undefined` and so the first `if` will never be entered.

Also, it is common to call these something with "handle" in their name, e.g. `clearKeysSoFarHandle`. This variable should be declared beside the `keysSoFar` variable to work as intended.

The `bind` call is not needed.

Formatting is off.

```js
var clearKeysSoFarHandle = null;
var keysSoFar = "";

// ...

function clearKeysSoFarAfterDelay() {
  // Restart delay if ongoing.
  if(clearKeysSoFarHandle != null) {
    clearTimeout(clearKeysSoFarHandle);
  }

  clearKeysSoFarHandle = setTimeout(function() {
    keysSoFar = "";
    clearKeysSoFarAfterDelay = null;
  }, 500);
}
```

Reply:
![image](https://user-images.githubusercontent.com/108732040/231361630-82172553-a3fc-4fef-9c8b-030b83f00c92.png)


After giving a though, we actually have the find feature separately. Hence removing as this is not needed.